java/arcs/sdk/android/storage/service/StorageService.kt

```diff
@@ -117,15 +122,81 @@ class StorageService : ResurrectorService() {
 
         writer.println()
 
+        if (DatabaseDriverProvider.isConfigured) {
+            val databaseManager = DatabaseDriverProvider.manager
+            val statistics = runBlocking { databaseManager.snapshotStatistics() }
+
+            if (statistics.isNotEmpty()) {
+                writer.println(
+                    """
+                        Databases:
+                        ----------
+                    """.trimIndent()
+                )
+            }
+
+            statistics.forEach { (identifier, snapshot) ->
+                val persistenceLabel = if (identifier.persistent) "persistent" else "non-persistent"
+                writer.println(
+                    """
+                        |  ${identifier.name} ($persistenceLabel):
+                    """.trimMargin("|")
+                )
+                snapshot.insertUpdate.dump(writer, pad = "    ", title = "Insertions/Updates")
+                snapshot.get.dump(writer, pad = "    ", title = "Gets")
+                snapshot.delete.dump(writer, pad = "    ", title = "Deletions")
+                writer.println()
+            }
+        }
+
         dumpRegistrations(writer)
     }
 
+    private fun PerformanceStatistics.Snapshot.dump(
```

Comment:
Would you like to consider adding an argument which resets snapshots per dump?
It's a nice-to-have feature that provides:
a) knowing the trend of periodical stats instead of historical one
b) constrain the counters' meaning only to my features or region-of-interests.

Reply:
Good idea. I'll make a bug for this.